File: src/olcw78 (die yoong)/jsAlgorithmStudy/week01/04_모의고사.go
package main

import "fmt"

func solution(answers []int) []int {
	s1 := []int{1, 2, 3, 4, 5}
	s2 := []int{2, 1, 2, 3, 2, 4, 2, 5}
	s3 := []int{3, 3, 1, 1, 2, 2, 4, 4, 5, 5}

	var m1, m2, m3 int = 0, 0, 0
	for i := 0; i < len(answers); i++ {
		cur := answers[i]
		if s1[i%len(s1)] == cur {
			m1++
		}

		if s2[i%len(s2)] == cur {
			m2++
		}

		if s3[i%len(s3)] == cur {
			m3++
		}
	}

	answer := make([]int, 0, 3)
	maxScore := max(m1, m2, m3)
	if m1 == maxScore {
		answer = append(answer, 1)
	}
	if m2 == maxScore {
		answer = append(answer, 2)
	}
	if m3 == maxScore {
		answer = append(answer, 3)
	}

	return answer
}

func main() {
	r1 := solution([]int{1, 2, 3, 4, 5})
	fmt.Println(r1)

	r2 := solution([]int{1, 3, 2, 4, 2})
	fmt.Println(r2)
}

File: src/olcw78 (die yoong)/jsAlgorithmStudy/week01/04_모의고사.js
/**
 * @param answers {number[]}
 */
function solution(answers) {
    const s1 = [1, 2, 3, 4, 5];
    const s2 = [2, 1, 2, 3, 2, 4, 2, 5];
    const s3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];

    let match1 = 0, match2 = 0, match3 = 0;

    for (let i = 0; i < answers.length; i++) {
        const cur = answers[i];
        if (s1[i % s1.length] === cur) match1++;
        if (s2[i % s2.length] === cur) match2++;
        if (s3[i % s3.length] === cur) match3++;
    }

    const answer = [];
    const maxScore = Math.max(match1, match2, match3);
    if (maxScore === match1) answer.push(1);
    if (maxScore === match2) answer.push(2);
    if (maxScore === match3) answer.push(3);

    return answer;
}

const r1 = solution([1, 2, 3, 4, 5])
console.log(r1)

const r2 = solution([1, 3, 2, 4, 2])
console.log(r2)

